src/shell/shell_cmds.cpp

```diff
@@ -625,13 +625,21 @@ void DOS_Shell::CMD_DIR(char * args) {
 	}
 
 	// Helper function to handle 'Press any key to continue' message
-	// regardless of specific formatting below.
-	// Call it whenever a newline gets printed.
+	// regardless of user-selected formatting of DIR command output.
 	//
-	// TODO: DIR code assumes, that terminal size is 80x25
+	// Call it whenever a newline gets printed to potentially display
+	// this one-line message.
+	//
+	// For some strange reason number of columns stored in BIOS segment
+	// is exact, while number of rows is 0-based (so 80x25 mode is
+	// represented as 80x24).  It's convenient for us, as it means we can
+	// get away with (p_count % term_height) instead of
+	// (p_count % (term_height - 1)).
+	//
+	const int term_height = real_readb(BIOSMEM_SEG, BIOSMEM_NB_ROWS);
 	auto show_press_any_key = [&]() {
 		p_count += 1;
-		if (optP && (p_count % 24) == 0)
+		if (optP && (p_count % term_height) == 0)
```

Comment:
Minor suggestion - the comment mentions "number of columns" and "number of rows", which is perfect and very readable; but the variable name dispenses with those unit names and stores the value as `height`. (I now wonder.. is `height` pixels, and we're going to convert to rows somewhere else?) 

Suggest aligning the comment and variable units and call it `term_rows`.

Reply:
I had `trows` in here originally, then I changed it to `term_height` to be in line with `term_width` from function `calc_column_widths`, but it won't hurt if we'll have the local variable names appropriate to context od specific functions :)

Changed to `term_rows`.